src/utils/results-processing.ts

```diff
@@ -1,9 +1,51 @@
-export default function resultsProcessing(cqlResults: object): any {
-  // TODO: Process FHIR resources returned from CQL Execution
+import { R4 } from '@ahryman40k/ts-fhir-types';
 
-  // Get each nonempty result
-  // Get name of expression
-  // Match with linkID of questionnaire
+// Filter results by querying proper data from the returned FHIR resources.
+// Modify the answerOptions of the questionnaire to include the results from execution
+export default function questionnaireUpdater(
+  cqlResults: any,
+  questionnaire: R4.IQuestionnaire,
+  patientId: string
+): R4.IQuestionnaire {
+  // Object Containing Questionnaire Items
+  let questionnaireItems: any = questionnaire.item;
 
-  return null;
+  // Get Non-Empty Patient Results
+  const igResources = cqlResults.patientResults[patientId];
+  for (let key in igResources) {
+    let resourceList = igResources[key];
+    if (resourceList.length > 0) {
+      // Find corresponding quesionnaire resource
+      const matchingResource = questionnaireItems.find((element: any) => element.linkId === key);
```

Comment:
```suggestion
      const matchingResource = questionnaireItems.find(element => element.linkId === key);
```
You should be able to drop the `any` type from this if you drop the any from the questionnaireItems variable above